qcodes/instrument/group_parameter.py

```diff
@@ -239,15 +239,20 @@ def _set_from_dict(self, calling_dict: Dict[str, Any]) -> None:
             raise RuntimeError("Trying to set GroupParameter not attached "
                                "to any instrument.")
         self.instrument.write(command_str)
+        for name, parameter in self.parameters.items():
+            parameter.cache.set(parameter._from_raw_value_to_value(
+                calling_dict[name]))
 
     def update(self) -> None:
         """
         Update the values of all the parameters within the group by calling
         the ``get_cmd``.
         """
-        if self.instrument is None:
-            raise RuntimeError("Trying to update GroupParameter not attached "
-                               "to any instrument.")
-        ret = self.get_parser(self.instrument.ask(self.get_cmd))
-        for name, p in list(self.parameters.items()):
-            p.get(parameter_value=ret[name])
+        if self.get_cmd is not None:
```

Comment:
It seems dangerous that this will silently do nothing if self.get_cmd is none. Could you elaborate on how this should work?

Reply:
The get_raw of the parameter first `updates` all the parameter of the group and then outputs the updated raw_value from the updated cache. In case the parameter does not have a get_cmd the GroupParameter was failing. With the above implementation if the get_cmd is None the GroupParameter falls back to taking cached value.

Reply:
Had a discussion offline. Agreed to revert it back as this seems dangerous to both of us.